jerry-core/include/jerryscript-core.h

```diff
@@ -458,9 +458,8 @@ jerry_value_t jerry_create_number_nan (void);
 jerry_value_t jerry_create_null (void);
 jerry_value_t jerry_create_object (void);
 jerry_value_t jerry_create_promise (void);
-jerry_value_t jerry_create_regexp (const jerry_char_t *pattern, jerry_regexp_flags_t flags);
-jerry_value_t jerry_create_regexp_sz (const jerry_char_t *pattern, jerry_size_t pattern_size,
-                                      jerry_regexp_flags_t flags);
+jerry_value_t jerry_create_regexp (const jerry_char_t *pattern, uint16_t flags);
```

Comment:
Is there a reasion why this is limited to 16 bits? Usually this takes a register regardless of type. I would use a 32 bit value.

Reply:
We could use a 32 bit value, however that would go way out of the scope of this patch.
In that case we must change everything related to `ecma_op_create_regexp_object`. I would like to leave it as it is in this patch, and create a follow up, that changes every related function's parameters to 32 bit values.

Reply:
I think @DanielBallaSZTE is right here. The API mirrors the internals.

On the other hand, if the goal is to break away from the internals, then I'd suggest _not_ to use a bitwidth-fixed (e.g., 32 bit) type, but a simple (`unsigned`?) `int` instead. Actually, that should be the fastest and easiest and most future-proof type on any machine.

Reply:
@akosthekiss Using an `unsigned int` requires you to cast it to `unsigned short int` (`uint16_t`) when calling the `ecma_op_create_regexp_object`. I already have a patch prepared that changes all the internal parameters to `uint32_t`, so I would like to leave it as an `uint16_t` for now and submit the patch if this one lands.

Reply:
Actually, I was not pushing for a change here. I was only reflecting to @zherczeg 's proposal to change this to `uint32_t`. And my comment was that _if_ we wanted a change (i.e., to have an API type that was different from the internal type) _then_ we should go for a non-sized type.